impl/src/main/java/org/jboss/weld/event/ObserverNotifier.java

```diff
@@ -347,9 +354,35 @@ public RuntimeException apply(Type eventType) {
             }), executor);
         }
 
-        // TODO if timeout set then wrap the completableFuture
+        // If NotificationOptionKeys.TIMEOUT is set, we will trigger the counter and use CompletableFuture.anyOf()
+        if (options.get(NotificationOptionKeys.TIMEOUT) != null) {
+            // TODO - any beter way to do the cast?
+            return new AsyncEventDeliveryStage<>((CompletableFuture<U>)CompletableFuture.anyOf(completableFuture, initiateTimer(options)), executor);
```

Comment:
This is still `TODO` but I don't know how to improve it. 
The cast itself should be harmless though - it only results to `CompletableFuture<Object>` because of the return value of `CompletableFuture.anyOf()`